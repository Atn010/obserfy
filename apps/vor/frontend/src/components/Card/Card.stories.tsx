import React, { FC } from "react"

import Card from "./Card"
import Typography from "../Typography/Typography"
import { generateFluidObject } from "../../__mocks__/data"
import { Image } from "../Image/Image"

export default {
  title: "Core|Card",
  component: Card,
  parameters: {
    componentSubtitle: "Just a simple Card",
  },
}

export const Basic: FC = () => (
  <Card m={2}>
    <Image height={400} src={generateFluidObject().src} />
    <Typography.H4 mx={4} mt={4}>
      Par example
    </Typography.H4>
    <Typography.Body m={4}>
      Lorem Ipsum is simply dummy text of the printing and typesetting industry.
      Lorem Ipsum has been the industry standard dummy text ever since the
      1500s, when an unknown printer took a galley of type and scrambled it to
      make a type specimen book.
    </Typography.Body>
  </Card>
)
