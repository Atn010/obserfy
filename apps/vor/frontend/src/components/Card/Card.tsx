import React, { FC } from "react"
import { BoxProps, Card as BaseCard } from "rebass"

export interface CardProps extends Omit<BoxProps, "css"> {
  borderRadius?: string | number | Array<string | number>
}
export const Card: FC<CardProps> = ({
  sx,
  borderRadius,
  maxWidth = "",
  ...props
}) => {
  return (
    <BaseCard
      p={0}
      maxWidth={maxWidth}
      overflow="hidden"
      sx={{
        ...sx,
        borderRadius,
      }}
      {...props}
    />
  )
}

export default Card
