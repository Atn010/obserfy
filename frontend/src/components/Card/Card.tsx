import React, { FC } from "react"
import { Card as BaseCard, BoxProps } from "rebass"

export interface CardProps extends Omit<BoxProps, "css"> {
  borderRadius?: string | number | Array<string | number>
}
export const Card: FC<CardProps> = ({
  sx,
  borderRadius,
  maxWidth = "",
  ...props
}) => {
  let sxStyle = sx
  if (borderRadius !== undefined) {
    sxStyle = Object.assign(sxStyle || {}, {
      borderRadius
    })
  }
  return (
    <BaseCard
      p={0}
      backgroundColor="surface"
      maxWidth={maxWidth}
      overflow="hidden"
      sx={sxStyle}
      {...props}
    />
  )
}

export default Card
