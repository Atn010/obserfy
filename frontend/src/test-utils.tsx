/* eslint-disable import/no-extraneous-dependencies */
import React, { FC, ReactElement } from "react"
import { ThemeProvider } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import { IntlProvider } from "gatsby-plugin-intl3"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import Theme from "./gatsby-plugin-theme-ui"
import Box from "./components/Box/Box"
import en from "./intl/en.json"

const AllProviders: FC = ({ children }) => {
  // TODO: Maybe this better be outside
  const history = createHistory(createMemorySource("/"))
  return (
    <LocationProvider history={history}>
      <ThemeProvider theme={Theme}>
        <IntlProvider locale="en" messages={en}>
          <Box fontSize={[16, 20]}>{children}</Box>
        </IntlProvider>
      </ThemeProvider>
    </LocationProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
