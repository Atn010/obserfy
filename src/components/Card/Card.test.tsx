import React from "react"
import { render } from "../../test-utils"
import { Basic } from "./Card.stories"

describe("Card", () => {
  it("should render correctly", () => {
    const { container } = render(<Basic />)
    expect(container).toMatchSnapshot()
  })
})
