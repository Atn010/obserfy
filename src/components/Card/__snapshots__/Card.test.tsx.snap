// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly 1`] = `
.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width:40em) {
  .emotion-4 {
    font-size: 20px;
  }
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding: 8px;
  background-color: var(--theme-ui-colors-background,#F7F8FA);
  border-radius: 6px;
  box-shadow: rgba(0,0,0,0.1) 0px 0px 1px,rgba(0,0,0,0.27) 0px 2px 4px -2px;
  padding: 0;
  background-color: var(--theme-ui-colors-surface,#fff);
  overflow: hidden;
  margin: 8px;
}

.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  height: 400px;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 24px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: 300;
  line-height: heading;
  font-size: 3.125rem;
  line-height: 6rem;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  color: var(--theme-ui-colors-text,rgba(0,0,0,0.87));
  margin-left: 32px;
  margin-right: 32px;
  margin-top: 32px;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: rgba(0,0,0,0.88);
  font-size: 1.6rem;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: 400;
  line-height: 3rem;
  color: var(--theme-ui-colors-text,rgba(0,0,0,0.87));
  font-size: 16px;
  margin: 32px;
}

@media screen and (min-width:40em) {
  .emotion-2 {
    font-size: 14px;
  }
}

<div>
  <div
    class="emotion-4"
  >
    <div
      class="emotion-3"
    >
      <img
        class="emotion-0"
        src="https://lorempixel.com/500/500/people"
      />
      <h4
        class="emotion-1"
      >
        Par example
      </h4>
      <p
        class="emotion-2"
      >
        Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.
      </p>
    </div>
  </div>
</div>
`;
